reclass/datatypes/parameters.py

```diff
@@ -172,11 +173,7 @@ def has_unresolved_refs(self):
         return len(self._occurrences) > 0
 
     def interpolate(self):
-        while self.has_unresolved_refs():
-            # we could use a view here, but this is simple enough:
-            # _interpolate_inner removes references from the refs hash after
-            # processing them, so we cannot just iterate the dict
-            path, refvalue = self._occurrences.iteritems().next()
+        for path, refvalue in self._occurrences.copy().items():
```

Comment:
UGH two-thread comms bane of my existence. No, sorry I meant the removal of the while on line 175.